tests/functional/test_ecosystem.py

```diff
@@ -52,6 +53,26 @@ def test_encode_address(ethereum):
     assert actual == raw_address
 
 
+def test_encode_calldata(ethereum):
+    abi = MethodABI(
+        type="function",
+        name="callMe",
+        inputs=[
+            ABIType(name="a", type="bytes4"),
+            ABIType(name="b", type="address"),
+            ABIType(name="c", type="uint256"),
+            ABIType(name="d", type="bytes4[]"),
+        ],
+    )
+    address = "0xd8da6bf26964af9d7eed9e03e53415d37aa96045"
+    byte_array = ["0x456", "0x678"]
+    values = ("0x123", address, HexBytes(55), byte_array)
+
+    actual = ethereum.encode_calldata(abi, *values)
+    expected = [HexBytes("0x123"), address, 55, [HexBytes(x) for x in byte_array]]
+    assert actual == expected
```

Comment:
@Ninjagod1251 This test fails now but should pass if we get this working, but also feel free to ad just any of this as needed and we can also chat about this on Monday. Thanks